pkg/artifactory/resource_artifactory_virtual_debian_repository.go

```diff
@@ -29,12 +29,8 @@ func resourceArtifactoryDebianVirtualRepository() *schema.Resource {
 			MinItems: 0,
 			Computed: true,
 			Elem: &schema.Schema{
-				Type: schema.TypeString,
-				ValidateFunc: validation.StringInSlice([]string{
-					"bz2",
-					"lzma",
-					"xz",
-				}, false),
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validateTypeSetStringList([]string{"bz2", "lzma", "xz"}),
```

Comment:
mmm, is there a reason to not use `validation.ToDiagFunc(validation.StringInSlice()` ? We actually use it elsewhere. Also, I think someone, somewhere, implemented `inList(ab,c,d,)` @danielmkn 

My understanding is that you couldn't do set/list validation - But if that's not the case, then this functionality is already implemented

Reply:
Checked with @danielmkn  ... the function he used is for ValidateFunc and not ValidateDiagFunc.
We couldn't use validation.ToDiagFunc because this TypeSet is not honored for ValidateDiagFunc.
Ref: https://github.com/hashicorp/terraform-plugin-sdk/blob/0c056990cdf4bd80eed47036d2f87702e1ea66fa/helper/schema/schema.go#L262
` ValidateDiagFunc is honored only when the schema’s Type is set to TypeInt, TypeFloat, TypeString, TypeBool, or TypeMap. It is ignored for all other types.`

Reply:
Ok, well, as long as you're not reimplementing something

Reply:
I think it's better to use the old way, and not write more code, then try to use the `new` way and write a bunch of code